app/code/Scandiweb/Test/Setup/Patch/Data/AddSimpleProduct.php

```diff
@@ -3,96 +3,187 @@
-use Magento\Framework\Setup\Patch\DataPatchInterface;
-use Magento\Framework\Setup\ModuleDataSetupInterface;
-use Magento\Catalog\Model\ProductFactory;
-use Magento\Catalog\Model\CategoryFactory;
+use Magento\Catalog\Api\CategoryLinkManagementInterface;
+use Magento\Catalog\Api\Data\ProductInterfaceFactory;
+use Magento\Catalog\Api\Data\CategoryInterfaceFactory;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\Product\Attribute\Source\Status;
+use Magento\Catalog\Model\Product\Type;
+use Magento\Catalog\Model\Product\Visibility;
+use Magento\Eav\Setup\EavSetup;
 use Magento\Framework\App\State;
-
-
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Magento\Framework\Validation\ValidationException;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+use Magento\InventoryApi\Api\Data\SourceItemInterfaceFactory;
+use Magento\InventoryApi\Api\SourceItemsSaveInterface;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Create Migration product class
+ */
 class AddSimpleProduct implements DataPatchInterface
 {
     /**
-     * @var ModuleDataSetupInterface
+     * @var ProductInterfaceFactory
      */
-    private $moduleDataSetup;
+    protected ProductInterfaceFactory $productInterfaceFactory;
 
     /**
-     * @var ProductFactory
+     * @var CategoryInterfaceFactory
      */
-    private $productFactory;
+    protected CategoryInterfaceFactory $categoryInterfaceFactory;
 
     /**
-     * @var CategoryFactory
+     * @var ProductRepositoryInterface
      */
-    private $categoryFactory;
+    protected ProductRepositoryInterface $productRepository;
 
     /**
      * @var State
      */
-    private $appState;
+    protected State $appState;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    protected StoreManagerInterface $storeManager;
+
+    /**
+     * @var SourceItemInterfaceFactory
+     */
+    protected SourceItemInterfaceFactory $sourceItemFactory;
 
+    /**
+     * @var SourceItemsSaveInterface
+     */
+    protected SourceItemsSaveInterface $sourceItemsSaveInterface;
+
+    /**
+     * @var EavSetup
+     */
+    protected EavSetup $eavSetup;
+
+    /**
+     * @var array
+     */
+    protected array $sourceItems = [];
+
+    /**
+     * Migration patch constructor.
+     *
+     * @param ProductInterfaceFactory $productInterfaceFactory
+     * @param CategoryInterfaceFactory $categoryInterfaceFactory
+     * @param ProductRepositoryInterface $productRepository
+     * @param SourceItemInterfaceFactory $sourceItemFactory
+     * @param SourceItemsSaveInterface $sourceItemsSaveInterface
+     * @param State $appState
+     * @param StoreManagerInterface $storeManager
+	 * @param EavSetup $eavSetup
```

Comment:
Fix padding

Reply:
Please run the file through autoformatting / replace tab characters with 4 whitespaces. Please configure your IDE to always replace tab characters with 4 whitespaces.
Use shortcut for phpstorm `ctrl+alt+l`